corehq/apps/locations/bulk_management.py

```diff
@@ -809,12 +809,21 @@ def update_children(parent_stub):
 
         for child_stub in child_stubs:
             child = child_stub.db_object
-            child.parent = parent_location
             if child_stub.do_delete:
                 # keep track of to be deleted items to delete them in top-to-bottom order
                 to_be_deleted.append(child)
             elif child_stub.needs_save:
                 child.location_type = types_by_code.get(child_stub.location_type)
+                if parent_location:
+                    # refetch parent_location object so that mptt related fields are updated consistently,
+                    #   since we are saving top to bottom, parent_location would not have any pending
+                    #   saves, so this is the right point to refetch the object.
+                    child.parent = SQLLocation.objects.get(
+                        domain=parent_location.domain,
+                        site_code=parent_code
+                    )
```

Comment:
We are saving breadth-first. Even then, we have to refetch objects in upper levels, because when lower level objects are saved, mptt might update mptt-fields of upper level objects. This is also the reason why caching won't work, as the prefetched upper objects won't get the mptt updates that are necessary to keep mptt consistent.

The only efficient solution that I found out so far will need following two to happen
1. The mptt PR that supports bulk-save needs to be merged and we would need to upgrade to that mptt
2. Remove couch syncing. Otherwise we would need to write a bulk couch sync (I have also tried this in the beginning, but thought it's not worth)